pysat/instruments/pysat_testing.py

```diff
@@ -111,56 +126,66 @@ def load(fnames, tag=None, sat_id=None, sim_multi_file_right=False,
     # fake orbit number
-    fake_delta = date  - pysat.datetime(2008,1,1) 
+    fake_delta = date  - (test_dates[date_tag][sat_id]-pds.DateOffset(years=1))
     fake_uts_root = fake_delta.total_seconds()
 
-    data['orbit_num'] = ((fake_uts_root+num_array)/5820.).astype(int)
+    data['orbit_num'] = ((fake_uts_root + num_array) / 5820.0).astype(int)
     
     # create a fake longitude, resets every 6240 seconds
     # sat moves at 360/5820 deg/s, Earth rotates at 360/86400, takes extra time 
     # to go around full longitude
     long_uts_root = np.mod(time_delta.total_seconds(), 6240)
-    longitude = np.mod(long_uts_root+num_array, 6240)*(360./6240.)
+    longitude = np.mod(long_uts_root + num_array, 6240) * (360.0 / 6240.0)
     data['longitude'] = longitude
 
     # create latitude area for testing polar orbits
-    latitude = 90.*np.cos(np.mod(uts_root+num_array, 5820)*(2.*np.pi/5820.)) 
+    latitude = 90.0 * np.cos(np.mod(uts_root + num_array, 5820) *
+                             (2.0 * np.pi / 5820.0)) 
     data['latitude'] = latitude
     
     # do slt, 20 second offset from mlt
-    uts_root = np.mod(time_delta.total_seconds()+20, 5820)
-    data['slt'] = np.mod(uts_root+num_array, 5820)*(24./5820.)
+    uts_root = np.mod(time_delta.total_seconds() + 20, 5820)
+    data['slt'] = np.mod(uts_root + num_array, 5820) * (24.0 / 5820.0)
     
     # create some fake data to support testing of averaging routines
     mlt_int = data['mlt'].astype(int)
-    long_int = (data['longitude']/15.).astype(int)
+    long_int = (data['longitude'] / 15.0).astype(int)
     data['dummy1'] = mlt_int
     data['dummy2'] = long_int
-    data['dummy3'] = mlt_int + long_int*1000.
+    data['dummy3'] = mlt_int + long_int * 1000.0
     data['dummy4'] = num_array
-    data['string_dummy'] = ['test']*len(data)
+    data['string_dummy'] = ['test'] * len(data)
     data['unicode_dummy'] = [u'test'] * len(data)
     data['int8_dummy'] = np.ones(len(data), dtype=np.int8)
     data['int16_dummy'] = np.ones(len(data), dtype=np.int16)
     data['int32_dummy'] = np.ones(len(data), dtype=np.int32)
     data['int64_dummy'] = np.ones(len(data), dtype=np.int64)
     # print (data['string_dummy'])
     
-    index = pds.date_range(data_date, data_date+pds.DateOffset(seconds=num-1), freq='S')
+    index = pds.date_range(data_date, data_date + pds.DateOffset(seconds=num-1),
+                           freq='S')
     data.index=index[0:num]
     data.index.name = 'Epoch'
     return data, meta.copy()
 
 
 def list_files(tag=None, sat_id=None, data_path=None, format_str=None):
     """Produce a fake list of files spanning a year"""
-    
-    index = pds.date_range(pysat.datetime(2008,1,1), pysat.datetime(2010,12,31)) 
-    names = [ data_path+date.strftime('%Y-%m-%d')+'.nofile' for date in index]
+
+    # Determine the appropriate date range for the fake files
+    date_tag = '' if tag not in tags else tag
+    start = test_dates[date_tag][sat_id] - pds.DateOffset(years=1)
```

Comment:
Original code covered two years. Also added protection for leap years.

start = test_dates[date_tag][sat_id] - pds.DateOffset(years=1)
stop = test_dates[date_tag][sat_id] + pds.DateOffset(years=2) \
       - pds.DateOffset(days=1)
index = pds.date_range(start, stop)

Reply:
I mean three years :)

Reply:
fixed